Gateway/Validator/InstallmentValidator.php

```diff
@@ -77,7 +78,9 @@ public function validate(array $validationSubject)
         $isValid = true;
         $fails = [];
         $payment = $validationSubject['payment'];
-        $quote = $this->session->getQuote();
+        $quoteId = $payment->getOrderId();
+        //This validator also runs for other payments that don't necesarily have a quoteId
+        $quote = $quoteId?$this->quoteRepository->get($quoteId):false;
```

Comment:
Not sure if this is the best way to check if a payment contains quoteId. 

In my tests when placing the order this validate() function ran twice, once for sales_order_payment and once for sales_quote_payment event prefixes, one of those do not contain quoteId and a NoSuchEntityException is thrown. This line prevents that. Feedback would be appreciated.